src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,161 @@
+import { PeopleFilters } from '../components/PeopleFilters';
+import { Loader } from '../components/Loader';
+import { PeopleTable } from '../components/PeopleTable';
+import React, { useEffect, useState } from 'react';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { useSearchParams } from 'react-router-dom';
+
+type Filter = {
+  query: string | null;
+  sex: string | null;
+  centuries: number[];
+  sort: string | null;
+  order: string | null;
+};
+
+function getParent(name: string | null, people: Person[]): Person | undefined {
+  return people.find(p => p.name === name);
+}
+
+const getFilteredPeople = (
+  people: Person[],
+  { query, sex, centuries, sort, order }: Filter,
+) => {
+  const filteredPeople = [...people].filter(person => {
+    if (sex && sex !== person.sex) {
+      return false;
+    }
+
+    if (
+      query &&
+      !person.name.toLowerCase().includes(query) &&
+      !person.motherName?.toLowerCase().includes(query) &&
+      !person.fatherName?.toLowerCase().includes(query)
+    ) {
+      return false;
+    }
+
+    if (!!centuries.length) {
+      return centuries.some(century => {
+        const startYear = (century - 1) * 100 + 1;
+        const endYear = century * 100;
+
+        return person.born > startYear && person.born < endYear;
+      });
+    }
+
+    return true;
+  });
+
+  if (sort) {
+    filteredPeople.sort((a, b) => {
+      switch (sort) {
+        case 'name':
+          return a.name.localeCompare(b.name);
+
+        case 'sex':
+          return a.sex.localeCompare(b.sex);
+
```

Comment:
```suggestion
        case 'name':
        case 'sex':
          return a[sort].localeCompare(b[sort]);

```
you can group some cases